contracts/LotteryNFT.sol

```diff
@@ -1,38 +1,94 @@
-pragma solidity 0.6.12;
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.6.0 <0.8.0;
 pragma experimental ABIEncoderV2;
 
-import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
 import "@openzeppelin/contracts/utils/Counters.sol";
 import "@openzeppelin/contracts/access/Ownable.sol";
+import "@nomiclabs/buidler/console.sol";
+import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+
 
-contract LotteryNFT is ERC721, Ownable {
+contract LotteryNFT is ERC1155, Ownable {
     using Counters for Counters.Counter;
     Counters.Counter private _tokenIds;
+    
+    uint256
+        private constant LOTTERY_INDEX = 0x00000000000000000000000000000000000000000000000000000000000007FF;
+    
+    address payable public manager;
+    string public name = "Pancake Lottery Ticket";
+    string public symbol = "PLT";
 
     mapping (uint256 => uint8[4]) public lotteryInfo;
     mapping (uint256 => uint256) public lotteryAmount;
     mapping (uint256 => uint256) public issueIndex;
     mapping (uint256 => bool) public claimInfo;
 
-    constructor() public ERC721("Pancake Lottery Ticket", "PLT") {}
+
+    event Ticket(
+        address indexed _player,
+        uint256 indexed _id,
+        uint256 indexed _amount
+    );
+
+    event TicketBatch(
```

Comment:
@pancake-swap do we need/use these events? These only add gas cost and not relevant since ERC1155 already incorporates events for minting